code/rd-vanilla/tr_init.cpp

```diff
@@ -1186,16 +1186,28 @@ Workaround for ri.Printf's 1024 characters buffer limit.
 
 void R_PrintLongString(const char *string) {
 	char buffer[1024];
-	const char *p;
+	const char *p, *s;
 	int size = strlen(string);
+	int copySize, bufferSize;
 
+	bufferSize = sizeof(buffer);
 	p = string;
-	while(size > 0)
+	while (size > 0)
 	{
-		Q_strncpyz(buffer, p, sizeof (buffer) );
-		Com_Printf( "%s", buffer );
-		p += 1023;
-		size -= 1023;
+		s = p + bufferSize - 1;
```

Comment:
This line of code is navigating 1023 bytes past the pointer p and then dereferencing that address on the next line. In effect this means that at the last iteration of the loop, it will **always** read memory past the end of the original string - thus a buffer over-read.